parse_test.go

```diff
@@ -191,6 +192,49 @@ func TestParseFile_SkipProcessingPixelDataValue(t *testing.T) {
 	})
 }
 
+func TestParseFile_AllowUnknownSpecificCharacterSet(t *testing.T) {
+	file, err := os.CreateTemp("", "unknown_specific_character_set.dcm")
+	if err != nil {
+		t.Fatalf("Unexpected error when creating tempfile: %v", err)
+	}
+
+	transferSyntaxUIDElement, err := dicom.NewElement(tag.TransferSyntaxUID, []string{uid.ImplicitVRLittleEndian})
+	if err != nil {
+		t.Fatalf("Unexpected error when creating transfer syntax uid element: %v", err)
+	}
+	specificCharacterSetElement, err := dicom.NewElement(tag.SpecificCharacterSet, []string{"UNKNOWN"})
+	if err != nil {
+		t.Fatalf("Unexpected error when creating specific character set element: %v", err)
+	}
+	patientNameElement, err := dicom.NewElement(tag.PatientName, []string{"Bob", "Jones"})
+	if err != nil {
+		t.Fatalf("Unexpected error when creating patient name element: %v", err)
+	}
+	unknownCharacterSetDataset := dicom.Dataset{Elements: []*dicom.Element{
+		transferSyntaxUIDElement,
+		specificCharacterSetElement,
+		patientNameElement,
+	}}
+	err = dicom.Write(file, unknownCharacterSetDataset)
+	if err != nil {
+		t.Errorf("Unexpected error writing dataset: %v", unknownCharacterSetDataset)
+	}
+	file.Close()
+
+	t.Run("WithoutAllowUnknownSpecificCharacterSet", func(t *testing.T) {
+		dataset, err := dicom.ParseFile(file.Name(), nil)
+		if err == nil {
```

Comment:
nit: might be better for us to inspect the error to ensure we're getting the right error (and not some other error due to future code changes) to ensure we're testing the right thing here, but not a big deal.